libs/widget-lib/src/IframeRpcProviderBridge.ts

```diff
@@ -0,0 +1,126 @@
+
+  /**
+   * Creates an instance of IframeRpcProviderBridge.
+   * @param iframeWidow - The iFrame window that will post up general RPC messages and to which the IframeRpcProviderBridge will forward the RPC result.
+   *  Also it will receive some special RPC events coming from the wallet, like connect/chainChanged,accountChanged
+   */
+  constructor(private iframeWidow: Window) {}
+
+  /**
+   * Disconnects the JSON-RPC bridge from the Ethereum provider.
+   */
+  disconnect() {
+    // Disconnect provider
+    this.ethereumProvider = null
+    window.removeEventListener('message', this.processRpcCallFromWindow)
+  }
+
+  /**
+   * Handles the 'connect' event and sets up event listeners for Ethereum provider events.
+   * @param ethereumProvider - The Ethereum provider to connect.
+   */
+  onConnect(ethereumProvider: EthereumProvider) {
+    if (this.ethereumProvider) {
+      this.disconnect()
+    }
+
+    // Save the provider
+    this.ethereumProvider = ethereumProvider
+
+    // Listen for messages coming to the main window (from the iFrame window)
+    window.addEventListener('message', this.processRpcCallFromWindow)
+
+    // Process pending requests
+    this.processPendingRequests()
+
+    // Register in the provider, the events that needs to be forwarded to the iFrame window
+    EVENTS_TO_FORWARD_TO_IFRAME.forEach((event) => {
+      ethereumProvider.on(event, this.onRpcEventForwardToIframe)
+    })
+  }
+
+  private processPendingRequests() {
+    // Process pending requests
+    Object.keys(this.requestWaitingForConnection).forEach((key) => {
+      this.processRequest(this.requestWaitingForConnection[key])
+    })
+
+    // Clear pending requests
+    this.requestWaitingForConnection = {}
+  }
+
+  /**
+   * Processes a JSON-RPC request and sends appropriate response or error via the content window.
+   * @param request - The JSON-RPC request to be processed.
+   */
+  processRequest(request: JsonRpcRequest) {
+    if (!this.ethereumProvider) return
+
+    const requestPromise =
+      request.method === 'enable' ? this.ethereumProvider.enable() : this.ethereumProvider.request(request)
+
+    // Do request, and forward the result or error to the iFrame window
+    requestPromise
+      .then((result) => this.forwardRpcCallToIframe({ id: request.id, result }))
+      .catch((error) => this.forwardRpcCallToIframe({ id: request.id, error }))
+  }
+
+  private processRpcCallFromWindow = (event: MessageEvent): void => {
+    if (!this.isRpcCall(event)) {
+      return
+    }
+
+    // If disconnected, ignore the message. Also persist the request
+    if (!this.ethereumProvider) {
+      if (event.data.id) {
+        this.requestWaitingForConnection[event.data.id] = event.data
+      }
+      return
+    }
+
+    // Delegate the rpc call
+    this.processRequest(event.data)
+  }
+
+  private isRpcCall(event: MessageEvent) {
+    return event.data.jsonrpc === '2.0'
```

Comment:
Is that what the const is for?

```suggestion
    return event.data.jsonrpc === JSON_PRC_V
```

Reply:
a339964e4f9c55495802af3aa1278ef8e79f26ba